runtime/dart_controller.cc

```diff
@@ -105,6 +105,13 @@ bool DartController::SendStartMessage(Dart_Handle root_library) {
   return LogIfError(result);
 }
 
+void DartController::RunFromKernel(const uint8_t* buffer, size_t size) {
+  tonic::DartState::Scope scope(dart_state());
+  LogIfError(Dart_LoadKernel(Dart_ReadKernelBinary(buffer, size)));
+  if (SendStartMessage(Dart_RootLibrary()))
+    exit(1);
```

Comment:
We are trying not to call `exit` from within the engine. On shells on the desktop, the proper exit code is required since it used by to tools. I think this is controlled via returning the proper `tonic::DartErrorHandleType` (@jason-simmons knows more about this mechanism and can comment here). On the mobile shells (especially iOS), applications must never call `exit`. I realize that there are other instances in this file that do call `exit`, but we are trying get rid of those too.